gene_benchmark/descriptor.py

```diff
@@ -648,3 +699,48 @@ def has_missing_columns(df_row, column_names):
 
     """
     return any(missing_col_or_nan(df_row, col) for col in column_names)
+
+
+class BasePairDescriptor(SingleEntityTypeDescriptor):
+    """A descriptor designated to describe each symbol by its base pair sequence."""
+
+    def __init__(self, specie: str = "human", description_col="description") -> None:
+        """
+        Initiates a base pair description object.
+
+        Args:
+        ----
+            specie (str, optional): The species to be used for the base pair sequence. Defaults to "human".
+            description_col (str, optional): Name of the description column. Defaults to "description".
+
+        """
+        self.species = specie
+        self.description_col = description_col
+        self.missing_entities = None
+        self.allow_missing = None
+        self.allow_partial = True
+
+    def _retrieve_dataframe_for_entities(
+        self, entities: list, first_description_only=False
+    ):
+        ensembles = _gene_symbol_to_ensemble_ids(entities, species=self.species)
+        base_pairs = {
+            symbol: _fetch_ensembl_sequence(ensemble)
+            for symbol, ensemble in ensembles.items()
+        }
+        pair_bse_df = pd.DataFrame(index=entities, columns=[self.description_col])
+        pair_bse_df[self.description_col] = [
+            base_pairs[v] if v in base_pairs else None for v in pair_bse_df.index
+        ]
+        return pair_bse_df
+
+    def get_missing_entities(self, elements, element_metadata_df):
+        return list(
+            filter(lambda x: x not in element_metadata_df.dropna().index, elements)
+        )
+
+    def row_to_description(self, df_row: pd.Series) -> str:
+        return df_row[self.description_col]
+
+    def is_partial_description_row(self, df_row: pd.Series) -> bool:
+        return False
```

Comment:
why does this function always return false?

Reply:
Because we don't have partial description here